doc/dev/osd_internals/manifest.rst

```diff
@@ -418,70 +422,53 @@ Operations:
 
   This operation is not a user mutation and does not trigger a clone to be created.
 
-  TODO: SET_CHUNK appears to clear the manifest unconditionally if it's not chunked.
-  That seems wrong. ::
+  TODO: SET_CHUNK appears to clear the manifest unconditionally if it's not chunked. ::
 
        if (!oi.manifest.is_chunked()) {
          oi.manifest.clear();
        }
 
-* tier-promote 
-
-  promote the object (including chunks). ::
+* evict-chunk
 
-        void tier_promote();
-
-        rados -p base_pool tier-promote <obj-name> 
+  Clears an extent from an object leaving only the manifest link between
+  it and the target_object. ::
 
-  Returns ENOENT if the object does not exist
-  Returns EINVAL if the object already is a redirect.
+        void evict_chunk(
+	  uint64_t offset, uint64_t length, int flag = 0);
+  
+        rados -p base_pool evict-chunk <offset> <length> <object>
 
-  For a chunked manifest, copies all chunks to head.
+  Note: this does not exist yet.
 
-  For a redirect manifest, copies data to head.
+(QUESTION: As I mentioned above, I think including this makes building a tiering
+agent far simpler)
 
-  TODO: To atomically replace a redirect or dedup'd chunk with a local copy atomically,
-  redirect will be clear after the promote.
+* tier-promote 
 
-  Does not clear the manifest.
+  promotes the object ensuring that subsequent reads and writes will be local ::
 
-  Note: For a chunked manifest, calls start_copy on itself and uses the
-  existing read proxy machinery to proxy the reads.
+        void tier_promote();
 
-  TODO: Use TIER_DO_NOT_DEDUP_PIN to avoid unnecessary dedup work.
-  - Two use cases.
+        rados -p base_pool tier-promote <obj-name> 
 
-    Case a: 
+  Returns ENOENT if the object does not exist
 
-      1. Create Object A and B
-      2. Setchunk A to B
-      3. Write A
-      4. TIER_DO_NOT_DEDUP_PIN
-      5. Flush does not occur
+  For a redirect manifest, copies data to head.
 
-    Case b:
+  TODO: Promote on a redirect object needs to clear the redirect.
 
-      1. Create Object A and B
-      2. Setchunk A to B
-      3. TIER_DO_NOT_DEDUP_PIN
-      4. Promote A
-      5. Write A
-      6. Flush does not occur
+  For a chunked manifest, reads all MISSING extents into the base pool, subsequ
```

Comment:
For a chunked manifest, reads all MISSING extents into the base pool, subsequ
 ????

Reply:
Oops, will fix.